server/internal/graphql/resolver/catalog_content.go

```diff
@@ -9,6 +9,22 @@ import (
 	"github.com/layer5io/meshery/server/models"
 )
 
+// CatalogPatternPage - represents a page of meshery patterns
+type catalogPatternPage struct {
+	Page       int               `json:"page"`
+	PageSize   int               `json:"page_size"`
+	TotalCount int               `json:"total_count"`
+	Patterns   []*model.CatalogPattern `json:"patterns"`
+}
+
+// CatalogFilterPage - represents a page of meshery filters
+type catalogFilterPage struct {
+	Page       int              `json:"page"`
+	PageSize   int              `json:"page_size"`
+	TotalCount int              `json:"total_count"`
+	Filters    []*model.CatalogFilter `json:"filters"`
+}
+
 func (r *queryResolver) fetchCatalogPattern(ctx context.Context, provider models.Provider, selector *model.CatalogSelector) ([]*model.CatalogPattern, error) {
 	token := ctx.Value(models.TokenCtxKey).(string)
 	resp, err := provider.GetCatalogMesheryPatterns(token, selector.Search, selector.Order)
```

Comment:
@abdullah1308 
We would have to add support for page and pagesize too...

So that UI request paginated result